src/renderers/shared/fiber/ReactFiber.js

```diff
@@ -338,10 +338,16 @@ function createFiberFromElementType(
 ) : Fiber {
   let fiber;
   if (typeof type === 'function') {
-    fiber = shouldConstruct(type) ?
-      createFiber(ClassComponent, key) :
-      createFiber(IndeterminateComponent, key);
-    fiber.type = type;
+    // @TODO (bvaughn) This is temporary hack just to get things working.
+    if (typeof type.__reactInternalHostComponentFlag !== 'undefined') {
+      fiber = createFiber(HostComponent, key);
```

Comment:
I hope this doesn't stay for a couple of years like similar hacks have stayed in Stack :-)